### ISSUE TITLE
WriteTo picks unexpected overload for the Console sink

### ISSUE THREAD

**Issue by lavrentiygg** [serilog/serilog-settings-configuration#318]
Consider the config section
```
        "WriteTo": [
            {
                "Name": "Console",
                "Args": {
                    "formatter": "Serilog.Formatting.Compact.RenderedCompactJsonFormatter, Serilog.Formatting.Compact"
                }
            }
        ]
```

The expectation is that "formatter" argument leads to `Console` method overload with `ITextFormatter formatter` parameter https://github.com/serilog/serilog-sinks-console/blob/dev/src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs#L98 - crucially there are 6 parameters: 1 sink, 1 formatter, 4 with default values provided

In reality, another overload is selected because it has 9 parameters: 1 sink, 8 with default values one of which has a type string

The reason is that `SelectConfigurationMethod` orders them based on number of parameters matching despite the fact that there is no `formatter` parameter in overload selected.

#234 most likely is a consequence of this

**Comment by nblumhardt** [comment]
Hi! Did you mean to close this? Thanks!

**Comment by lavrentiygg** [comment]
Yes, sorry it was more convoluted than I thought initially. If it proves to be a issue, I'll reopen. Thanks for the follow up!
